src/pages/filter/ui/filterAttributes.store.ts

```diff
@@ -41,23 +39,36 @@ export class FilterAttributesStore {
     makeAutoObservable(this)
   }
 
+  public setCurrentMode(modeType?: ModeTypes): void {
+    this.currentMode = modeType ?? undefined
+  }
+
+  public resetCurrentMode(): void {
+    this.currentMode = undefined
+  }
+
   public get currentGroup(): FilterGroup {
     return {
       vgroup: this.filterStore.selectedGroupItem.vgroup,
       groupName: this.filterStore.selectedGroupItem.name,
     }
   }
 
-  public get groupSubKind(): string {
-    return this.filterStore.selectedGroupItem['sub-kind']
+  get filteredEnumVariants(): [string, number][] {
+    return toJS(this.getAllEnumVariants(this.currentGroup))
   }
 
-  public get allEnumVariants(): [string, number][] {
-    return toJS(this.getAllEnumVariants(this.currentGroup))
+  get allEnumVariants(): [string, number][] {
+    const allEnumVariants: [string, number][] =
+      this.datasetStore.dsStat['stat-list']?.find(
+        (item: any) => item.name === this.currentGroup.groupName,
+      )?.variants ?? []
+
+    return toJS(allEnumVariants)
   }
 
-  public get datasetEnumValues(): string[] {
-    return toJS(this.getDatasetEnumValues(this.currentGroup))
+  public get groupSubKind(): string {
+    return this.filterStore.selectedGroupItem['sub-kind']
```

Comment:
I know it's not yours modifications but please remove all dependencies from the constructor. No `this.filterStore` should be here.